editoast/src/views/timetable/similar_trains.rs

```diff
@@ -815,4 +864,219 @@ mod tests {
+            .core_client(core.into())
+            .build();
+        let small_infra = create_small_infra(&mut db_pool.get_ok()).await;
+        let timetable = create_timetable(&mut db_pool.get_ok()).await;
+        let rolling_stock =
+            create_fast_rolling_stock(&mut db_pool.get_ok(), &Uuid::new_v4().to_string()).await;
+
+        let train_name = "train_1".to_string();
+        let path: Vec<PathItem> = vec![
+            PathItem::new_operational_point("West_station"), // WS
+            PathItem::new_operational_point("Mid_West_station"), // MWS
+            PathItem::new_operational_point("Mid_East_station"), // MES
+            PathItem::new_operational_point("North_station"), // NS
+            PathItem::new_operational_point("South_station"), // SS
+        ];
+        let schedule: Vec<ScheduleItem> = vec![
+            ScheduleItem::new_with_stop(
+                "Mid_West_station",
+                Duration::new(300, 0).expect("Failed to parse duration"),
+            ),
+            ScheduleItem::new_with_stop(
+                "North_station",
+                Duration::new(300, 0).expect("Failed to parse duration"),
+            ),
+            ScheduleItem::new_with_stop(
+                "South_station",
+                Duration::new(0, 0).expect("Failed to parse duration"),
+            ),
+        ];
+        let start_time =
+            DateTime::from_str("2025-01-01T10:00:00Z").expect("Failed to parse datetime");
+
+        // WS(22):stop  MWS(33):stop  MES(44):passing_by  NS(55):stop  SS(66):stop
+        create_train_schedule(
+            &mut db_pool.get_ok(),
+            timetable.id,
+            train_name.clone(),
+            rolling_stock.name.clone(),
+            path,
+            start_time,
+            schedule,
+        )
+        .await;
+
+        // MWS(33):stop  MES(44):passing_by  NS(55):stop
+        let request = Request {
+            rolling_stock: RollingStockCharacteristics {
+                name: rolling_stock.name.clone(),
+                speed_limit_tag: None,
+            },
+            waypoints: vec![
+                Waypoint::new(33, "BV".to_string(), true),
+                Waypoint::new(44, "BV".to_string(), false),
+                Waypoint::new(55, "BV".to_string(), true),
+            ],
+            infra_id: small_infra.id,
+            timetable_id: timetable.id,
+        };
+        let request = app.post("/similar_trains").json(&request);
+        let response: Response = app.fetch(request).assert_status(StatusCode::OK).json_into();
+
+        let expected_response = Response {
+            similar_trains: vec![SimilarTrainItem {
+                train_name: train_name.clone().into(),
+                start_time,
+                begin: WaypointResponse::new(33, "BV"),
+                end: WaypointResponse::new(55, "BV"),
+            }],
+        };
+        assert_eq!(response, expected_response);
+
+        // NS(55):stop SS(66):stop
```

Comment:
...and this look like a second test. Maybe use `#[rstest::case]` to make 2 tests out of it.